src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.Minimal.cs

```diff
@@ -50,14 +50,14 @@ public partial struct ImmutableArray<T> : IEnumerable<T>, IEquatable<ImmutableAr
         /// This would be private, but we make it internal so that our own extension methods can access it.
         /// </remarks>
         [DebuggerBrowsable(DebuggerBrowsableState.RootHidden)]
-        internal T[] array;
+        internal T[]? array;
```

Comment:
The array field itself should be non-nullable. ImmutableArray acts as a zero overhead wrapper, thus no member do null check. Use `! ` doesn't look correct.

Reply:
Your point makes sense, but as i have seen some methods checking null (using ThrowInvalidOperationIfNotInitialized()). So i assume throwing NRE is not intention for all case. Using `!` is expressing it could throw NRE and i am OK with that so suppress the warning. IMO that is what we want to do with nullabilty analysis.

Reply:
I'm ok either way.  As for when ThrowInvalidOperationIfNotInitialized is used, note the comment on `ThrowInvalidOperationIfNotInitialized`:
https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.Minimal.cs#L435